x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/template/IndexTemplateRegistryTests.java

```diff
@@ -397,15 +397,16 @@ public void testAutomaticRollover() throws Exception {
         assertBusy(() -> assertThat(rolloverCounter.get(), equalTo(2)));
         AtomicReference<Collection<RolloverResponse>> rolloverResponsesRef = registry.getRolloverResponses();
         assertBusy(() -> assertNotNull(rolloverResponsesRef.get()));
-        Collection<RolloverResponse> rolloverResponses = rolloverResponsesRef.get();
-        assertThat(rolloverResponses, hasSize(2));
+        assertThat(rolloverResponsesRef.get(), hasSize(2));
 
         // test again, to verify that the per-index-template creation lock gets released for reuse
         putIndexTemplateCounter.set(0);
         rolloverCounter.set(0);
         registry.clusterChanged(event);
+        rolloverResponsesRef.set(Collections.emptySet());
```

Comment:
Shouldn't setting it to an empty set happen before we call the `clusterChanged` event?

Reply:
Oh wow I thought I'd done that. Good catch!

Reply:
(although in practice that seemed to work in the couple of hundred iterations I tried where I put in a forced Thread.sleep in IndexTemplateRegistry)